src/domain/app/header/Header.tsx

```diff
@@ -56,10 +56,24 @@ const Header: React.FC = () => {
   const navigate = useNavigate();
   const location = useLocation();
   const { changeLanguage, languageOptions } = useSelectLanguage();
-  const { authenticated, login, logout } = useAuth();
+  const { authenticated, login, logout, user: authUser } = useAuth();
 
   const { t } = useTranslation();
   const { user } = useUser();
+  const [displayName, setDisplayName] = useState('');
+
+  useEffect(() => {
+    if (authenticated) {
+      if (authUser?.profile) {
+        const { name, family_name, given_name } = authUser?.profile;
+        if (name) setDisplayName(name);
+        else if (given_name && family_name) {
+          setDisplayName(`${given_name} ${family_name}`);
+        } else if (given_name) setDisplayName(given_name);
+        else setDisplayName('');
+      } else setDisplayName('');
+    }
+  }, [authUser, authenticated]);
```

Comment:
This change breaks unit tests in `src/domain/app/header/__tests__/Header.test.tsx` file.
You can fix failing tests by changing
```
import {
  mockedRegistrationUserResponse,
  mockedRegularUserResponse,
  mockedUserResponse,
  mockedUserWithoutOrganizationsResponse,
  userFirstName,
} from '../../../user/__mocks__/user';

...

const findUserMenuButton = () =>
  screen.findByRole('button', { name: userFirstName }, { timeout: 10000 });
``` 
to
```
import {
  mockedRegistrationUserResponse,
  mockedRegularUserResponse,
  mockedUserResponse,
  mockedUserWithoutOrganizationsResponse,
  userName,
} from '../../../user/__mocks__/user';

...

const findUserMenuButton = () =>
  screen.findByRole('button', { name: userName }, { timeout: 10000 });
```